astropy/timeseries/binned.py

```diff
@@ -119,6 +119,13 @@ def __init__(self, data=None, *, time_bin_start=None, time_bin_end=None,
         if isinstance(time_bin_size, TimeDelta):
             time_bin_size = time_bin_size.sec * u.s
 
+        # If any one of the parameters 'time_bin_start' or 'time_bin_end' is not scalar,
+        # an exception is raised.
+        if n_bins is not None and time_bin_end is not None:
+            if not (time_bin_start.isscalar and  time_bin_end.isscalar):
+                raise TypeError("'n_bins' is  not valid as 'time_bin_start' or 'time_bin_size'"
```

Comment:
The above comment applies here too.
```suggestion
                raise TypeError("'n_bins' is  not valid as 'time_bin_start' or"
                                             'time_bin_size' are not scalar.")
```